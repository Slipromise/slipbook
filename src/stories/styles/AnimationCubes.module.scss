.container {
    position: relative;
    // height: 500px;
    // width: 500px;

    transform: skewY(-20deg);

    .cube {
        position: relative;

        translate: calc(var(--z) * 60px) calc(var(--z) * 60px);

        >div {
            position: absolute;

            translate: calc(-70px * var(--x));



            >span {
                position: relative;
                display: inline-block;
                width: 50px;
                height: 50px;
                background-color: #dcdcdc;
                transition: 1.5s;

                &.active {
                    background-color: #ef4149;
                    transform: translate(0, -50px);
                    animation: colorAnimation var(--duration) ease-in-out infinite;

                    &::before {
                        background-color: #f75d64;
                    }

                    &::after {
                        background-color: #f14e55;
                        box-shadow: -150px 150px 5px rgba(0, 0, 0, 0.15);
                    }
                }

                &::before {
                    content: "";
                    position: absolute;
                    left: -40px;
                    width: 40px;
                    height: 100%;
                    background-color: #c8c8c8;
                    transform-origin: right;
                    transform: skewY(45deg);
                    transition: 1.5s;
                }

                &::after {
                    content: '';
                    position: absolute;
                    top: -40px;
                    left: 0;
                    width: 100%;
                    height: 40px;
                    background-color: #f2f2f2;
                    transform-origin: bottom;
                    transform: skewX(45deg);
                    transition: 1.5s;
                    box-shadow: -100px 100px 5px rgba(0, 0, 0, 0.15);
                }
            }

        }

    }
}

@keyframes colorAnimation {
    0% {
        filter: hue-rotate(0deg);
    }

    100% {
        filter: hue-rotate(360deg);
    }
}