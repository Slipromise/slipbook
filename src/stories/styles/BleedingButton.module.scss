.container{

    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: visible;
    text-decoration: none;

    .bubbles{
        position: absolute;
        background-color: #f00;
        width: 100%;
        height: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        border-radius: 1.7rem;
        filter: url(#gooey);
        
        .bubble{
            position: absolute;
            width: 20px;
            height: 20px;
            background: #f00;
            border-radius: 50%;
            pointer-events: none;
            animation: bubble 1.5s linear infinite;
            animation-delay: calc(var(--i) * 0.2s);
        }
    }
    .text{
        // position: absolute;
        z-index: 50;
        background-color: #111;
        color: red;
        margin: 0 3px;
        padding: 1rem;
        border-radius: 2rem;
        font-size: 1.5rem;
        font-weight: 500;
        text-transform: uppercase;
        transition: all 0.3s ease;
        
    }
    &:hover{
        .text{
            background-color: #f00;
            color: white;
            font-weight: 700;
            font-size: 2rem;
        }
    }
}

@keyframes bubble {
    0% {
        transform: translate(0) scale(1);
    }
    100% {
        transform: translate(var(--dx),var(--dy)) scale(0.5);
    }
}