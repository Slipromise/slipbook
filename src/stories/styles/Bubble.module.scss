@use "../../styles/mixins/bubble-random" as *;

.normal {
    width: 200px;
    height: 200px;
    background-color: red;
    @include bubble-random;
}

.stereoscopic {
    width: 200px;
    height: 200px;
    background: radial-gradient(circle at 0% 0%,
            rgb(3, 110, 250),
            rgb(5, 232, 164));
    box-shadow: 0 -2vmin 4vmin rgb(26, 81, 165) inset,
        0 1vmin 4vmin rgb(54, 131, 71) inset,
        0 -2vmin 7vmin rgb(66, 131, 121) inset;
    @include bubble-random;
}

.animation {
    animation-name: bubbleAni;
    animation-duration: 10s;
    animation-iteration-count: infinite;
}

@keyframes bubbleAni {
    $count: 7;
    $space: 100 / ($count + 1);

    @for $i from 1 through $count {
        #{$space * $i * 1% } {
            @include bubble-random;
        }
    }
}