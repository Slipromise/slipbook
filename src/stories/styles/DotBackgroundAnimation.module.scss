@use 'sass:math';

.container {
    position: absolute;
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    overflow: hidden;
    justify-content: center;

    .dot {
        width: 50px;
        height: 50px;
        border-radius: 50%;
        background-color: #94fe08;
        transform: scale(var(--scale,.5));
    }

}