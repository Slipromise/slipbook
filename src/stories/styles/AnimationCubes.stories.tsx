
import type { Meta, StoryObj } from "@storybook/react";
import React, { useState } from "react";
import styles from "./AnimationCubes.module.scss";
import { useInterval } from "react-use";
import { sample } from "lodash";

type Props = { updateRate?: number; duration?: number };

const CUBE_SIDE = [-3, -2, -1, 0, 1, 2, 3];

function AnimationCubes({
  updateRate = 500,
  duration = 2000,
}: Props) {
  const [activeIndexes, setActiveIndexes] = useState<string[]>(() => []);

  useInterval(
    () => {
      let activeIndex: string;

      setActiveIndexes((pre) => {
        activeIndex = `${sample(CUBE_SIDE)},${sample(CUBE_SIDE)}`;
        while (pre.includes(activeIndex)) {
          activeIndex = `${sample(CUBE_SIDE)},${sample(CUBE_SIDE)}`;
        }
        pre.push(activeIndex);
        return [...pre];
      });

      setTimeout(() => {
        setActiveIndexes((pre) => pre.filter((v) => v !== activeIndex));
      }, duration);
    },
    CUBE_SIDE.length * CUBE_SIDE.length > activeIndexes.length
      ? updateRate
      : undefined
  );

  return (
    <div
      className={styles.container}
      style={{ "--duration": `${duration}ms` } as React.CSSProperties}
    >
      {CUBE_SIDE.map((z, i) => (
        <div
          key={i}
          style={{ "--z": z } as React.CSSProperties}
          className={styles.cube}
        >
          {CUBE_SIDE.map((x, _i) => (
            <div style={{ "--x": x } as React.CSSProperties} key={_i}>
              <span
                className={
                  activeIndexes.includes(`${z},${x}`)
                    ? styles.active
                    : undefined
                }
              />
            </div>
          ))}
        </div>
      ))}
    </div>
  );
}

const meta: Meta<typeof AnimationCubes> = {
  component: AnimationCubes,
  parameters: {
    layout: "centered",
  },
};
export default meta;

type Story = StoryObj<typeof AnimationCubes>;

export const Normal: Story = {
  args: {
    updateRate: 500,
    duration: 2000,
  },
};

export const FastUpdates: Story = {
  args: {
    updateRate: 250,
    duration: 4000,
  },
};
