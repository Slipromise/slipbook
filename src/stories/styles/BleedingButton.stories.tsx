import type { Meta, StoryObj } from "@storybook/react";
import styles from "./BleedingButton.module.scss";
import { useMemo } from "react";

type Props = {
  particleCount?: number;
};

function BleedingButton({ particleCount = 100 }: Props) {
  const bubbles = useMemo(() => {
    const result = Array.from({ length: particleCount }).map((_, i) => {
      const angle = Math.random() * 360 * Math.PI;
      const distance = Math.random() * 100 + 150;

      return {
        "--dx": `${Math.cos(angle) * distance}px`,
        "--dy": `${Math.sin(angle) * distance}px`,
        "--i": Math.ceil((i + 1) / 3),
      } as React.CSSProperties;
    });
    return result;
  }, [particleCount]);

  return (
    <>
      <a className={styles.container}>
        <div className={styles.bubbles}>
          {bubbles.map((style, index) => (
            <div key={index} className={styles.bubble} style={style} />
          ))}
        </div>
        <span className={styles.text}>Bleeding</span>
      </a>
      <svg style={{ display: "none" }}>
        <filter id="gooey">
          <feGaussianBlur in="SourceGraphic" stdDeviation="10" />
          <feColorMatrix
            values="
        1 0 0 0 0
        0 1 0 0 0
        0 0 1 0 0
        0 0 0 50 -10"
          ></feColorMatrix>
        </filter>
      </svg>
    </>
  );
}

const meta: Meta<typeof BleedingButton> = {
  component: BleedingButton,
  parameters: {
    layout: "centered",
    backgrounds: { default: "dark" },
  },
};

export default meta;

type Story = StoryObj<typeof BleedingButton>;

export const Normal: Story = {
  args: {},
};

export const MoreParticles: Story = {
    args:{
        particleCount: 500,
    }
}