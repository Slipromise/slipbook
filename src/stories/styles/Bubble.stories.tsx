import type { Meta, StoryObj } from "@storybook/react";
import styles from "./Bubble.module.scss";

import React from "react";

type Props = {
  className?: React.HtmlHTMLAttributes<HTMLDivElement>["className"];
};

function Bubble({ className }: Props) {
  return <div className={className}></div>;
}

const meta: Meta<typeof Bubble> = {
  component: Bubble,
  parameters: {
    layout: "centered",
  },
};

export default meta;
type Story = StoryObj<typeof Bubble>;

export const Normal: Story = {
  args: {
    className: styles.normal,
  },
};

export const Stereoscopic: Story = {
  args: {
    className: styles.stereoscopic,
  },
};

export const Animation: Story = {
  args: {
    className: `${styles.stereoscopic} ${styles.animation}`,
  },
};
