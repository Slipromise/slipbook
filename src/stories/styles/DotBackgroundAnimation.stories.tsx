import type { Meta, StoryObj } from "@storybook/react";
import styles from "./DotBackgroundAnimation.module.scss";
import { CSSProperties, useEffect, useMemo, useRef } from "react";

function DotBackgroundAnimation() {
  const containerRef = useRef<HTMLDivElement>(null);

  const dots = useMemo(() => {
    const length =
      Math.ceil(window.innerWidth / 50) * Math.ceil(window.innerHeight / 50);
    return Array.from({ length }).fill("");
  }, []);

  useEffect(() => {
    const eventListener = (event: MouseEvent) => {
      containerRef.current?.style.setProperty(
        "--mouse-x",
        `${event.clientX}px`
      );
      containerRef.current?.style.setProperty(
        "--mouse-y",
        `${event.clientY}px`
      );

      containerRef.current
        ?.querySelectorAll(`.${styles.dot}`)
        .forEach((dot) => {
          const x = (dot as HTMLElement).offsetLeft - event.clientX + 25;
          const y = (dot as HTMLElement).offsetTop - event.clientY + 25;

          const dist = Math.sqrt(x * x + y * y);
          const scroe = Math.exp(dist * -0.01);
          (dot as HTMLElement).style.setProperty("--scale", `${scroe * 6}`);
        });
    };

    window.document.addEventListener("mousemove", eventListener);

    return () => {
      window.document.removeEventListener("mousemove", eventListener);
    };
  }, []);

  return (
    <div className={styles.container} ref={containerRef}>
      {dots.map((_, index) => (
        <div
          key={index}
          className={styles.dot}
          style={{ "--i": index } as CSSProperties}
        />
      ))}
    </div>
  );
}

const meta: Meta<typeof DotBackgroundAnimation> = {
  component: DotBackgroundAnimation,
  parameters: {
    layout: "fullscreen",
    backgrounds: {
      default: "dark",
    },
  },
};

export default meta;

type Story = StoryObj<typeof DotBackgroundAnimation>;

export const Default: Story = {};
