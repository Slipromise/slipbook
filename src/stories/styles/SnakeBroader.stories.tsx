import type { Meta, StoryObj } from "@storybook/react";
import styles from "./SnakeBroader.module.scss";

import React from "react";

type Props = {
  className?: React.HtmlHTMLAttributes<HTMLDivElement>["className"];
  borderSideClassName?: React.HtmlHTMLAttributes<HTMLDivElement>["className"];
};

function SnakeBroader({ className, borderSideClassName }: Props) {
  return (
    <div className={className}>
      <span className={borderSideClassName} />
      <span className={borderSideClassName} />
      <span className={borderSideClassName} />
      <span className={borderSideClassName} />
    </div>
  );
}

const meta: Meta<typeof SnakeBroader> = {
  component: SnakeBroader,
  parameters: {
    backgrounds: { default: "dark" },
  },
};

export default meta;

type Story = StoryObj<typeof SnakeBroader>;

export const Animation4SidesClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-4-sides-clockwise"],
  },
};

export const Animation4SidesCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-4-sides-counterclockwise"],
  },
};

export const Animation2SidesClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-2-sides-clockwise"],
  },
};

export const Animation2SidesCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-2-sides-counterclockwise"],
  },
};

export const AnimationSideClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-1-sides-clockwise"],
  },
};

export const AnimationSideCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["animation-1-sides-counterclockwise"],
  },
};

export const Hover4SidesClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-4-sides-clockwise"],
  },
};

export const Hover4SidesCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-4-sides-counterclockwise"],
  },
};

export const Hover2SidesClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-2-sides-clockwise"],
  },
};

export const Hover2SidesCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-2-sides-counterclockwise"],
  },
};

export const HoverSideClockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-1-sides-clockwise"],
  },
};

export const HoverSideCounterclockwise: Story = {
  args: {
    className: styles.container,
    borderSideClassName: styles["hover-1-sides-counterclockwise"],
  },
};
