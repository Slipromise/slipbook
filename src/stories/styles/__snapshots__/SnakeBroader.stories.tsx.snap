// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/styles/SnakeBroader Animation2SidesClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-2-sides-clockwise_1xvzd_168">
  </span>
  <span class="_animation-2-sides-clockwise_1xvzd_168">
  </span>
  <span class="_animation-2-sides-clockwise_1xvzd_168">
  </span>
  <span class="_animation-2-sides-clockwise_1xvzd_168">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Animation2SidesCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-2-sides-counterclockwise_1xvzd_252">
  </span>
  <span class="_animation-2-sides-counterclockwise_1xvzd_252">
  </span>
  <span class="_animation-2-sides-counterclockwise_1xvzd_252">
  </span>
  <span class="_animation-2-sides-counterclockwise_1xvzd_252">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Animation4SidesClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-4-sides-clockwise_1xvzd_8">
  </span>
  <span class="_animation-4-sides-clockwise_1xvzd_8">
  </span>
  <span class="_animation-4-sides-clockwise_1xvzd_8">
  </span>
  <span class="_animation-4-sides-clockwise_1xvzd_8">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Animation4SidesCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-4-sides-counterclockwise_1xvzd_88">
  </span>
  <span class="_animation-4-sides-counterclockwise_1xvzd_88">
  </span>
  <span class="_animation-4-sides-counterclockwise_1xvzd_88">
  </span>
  <span class="_animation-4-sides-counterclockwise_1xvzd_88">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader AnimationSideClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-1-sides-clockwise_1xvzd_336">
  </span>
  <span class="_animation-1-sides-clockwise_1xvzd_336">
  </span>
  <span class="_animation-1-sides-clockwise_1xvzd_336">
  </span>
  <span class="_animation-1-sides-clockwise_1xvzd_336">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader AnimationSideCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_animation-1-sides-counterclockwise_1xvzd_420">
  </span>
  <span class="_animation-1-sides-counterclockwise_1xvzd_420">
  </span>
  <span class="_animation-1-sides-counterclockwise_1xvzd_420">
  </span>
  <span class="_animation-1-sides-counterclockwise_1xvzd_420">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Hover2SidesClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-2-sides-clockwise_1xvzd_680">
  </span>
  <span class="_hover-2-sides-clockwise_1xvzd_680">
  </span>
  <span class="_hover-2-sides-clockwise_1xvzd_680">
  </span>
  <span class="_hover-2-sides-clockwise_1xvzd_680">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Hover2SidesCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-2-sides-counterclockwise_1xvzd_768">
  </span>
  <span class="_hover-2-sides-counterclockwise_1xvzd_768">
  </span>
  <span class="_hover-2-sides-counterclockwise_1xvzd_768">
  </span>
  <span class="_hover-2-sides-counterclockwise_1xvzd_768">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Hover4SidesClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-4-sides-clockwise_1xvzd_504">
  </span>
  <span class="_hover-4-sides-clockwise_1xvzd_504">
  </span>
  <span class="_hover-4-sides-clockwise_1xvzd_504">
  </span>
  <span class="_hover-4-sides-clockwise_1xvzd_504">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader Hover4SidesCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-4-sides-counterclockwise_1xvzd_592">
  </span>
  <span class="_hover-4-sides-counterclockwise_1xvzd_592">
  </span>
  <span class="_hover-4-sides-counterclockwise_1xvzd_592">
  </span>
  <span class="_hover-4-sides-counterclockwise_1xvzd_592">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader HoverSideClockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-1-sides-clockwise_1xvzd_856">
  </span>
  <span class="_hover-1-sides-clockwise_1xvzd_856">
  </span>
  <span class="_hover-1-sides-clockwise_1xvzd_856">
  </span>
  <span class="_hover-1-sides-clockwise_1xvzd_856">
  </span>
</div>
`;

exports[`stories/styles/SnakeBroader HoverSideCounterclockwise smoke-test 1`] = `
<div class="_container_1xvzd_1">
  <span class="_hover-1-sides-counterclockwise_1xvzd_944">
  </span>
  <span class="_hover-1-sides-counterclockwise_1xvzd_944">
  </span>
  <span class="_hover-1-sides-counterclockwise_1xvzd_944">
  </span>
  <span class="_hover-1-sides-counterclockwise_1xvzd_944">
  </span>
</div>
`;
